src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -600,28 +601,30 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
     }
 
     /**
-     * Finds and returns the first sentence.
+     * Finds the first sentence.
      *
      * @param ast The Javadoc root node.
      * @param period The configured period symbol.
-     * @return The first sentence up to and excluding the period, or null if no ending was found.
+     * @return An Optional containing the first sentence
+     *     up to and excluding the period, or an empty
+     *     Optional if no ending was found.
      */
-    @Nullable
-    private static String getFirstSentenceOrNull(DetailNode ast, String period) {
+    private static Optional<String> getFirstSentence(DetailNode ast, String period) {
         final List<String> sentenceParts = new ArrayList<>();
-        String sentence = null;
+        Optional<String> result = Optional.empty();
```

Comment:
Let's do refactoring step by step, in few PRs.
It will be easier for both sides.
If something can be done in next PR let's do it in next PR, we can keep issue open as much as we need.

This PR is  already good milestone. 
@mohitsatr, if some suggestions is easy let's do it in this PR, if not that easy, better to postpone to next PR.

Reply:
> why not return Optional.ofNullable(result) here?

similar to https://github.com/checkstyle/checkstyle/pull/16020#discussion_r1887845399

> simply stream the "text parts" and use the Stream#takeWhile API

alright. I'will look into it and try to refactor. 
yes, it'd better do it in separate PR.

Reply:
>why not `return Optional.ofNullable(result) ` here ?

To try avoid null if that doesn't not make code hard to read. Fashioned code. As whole this issue.

Reply:
attention to detail please, there is not `null` as special meaning in method internals.
We say, that default value of result is empty result.
in loop it is substituted to non-nullable value !!! attention to `Optional.of`
![image](https://github.com/user-attachments/assets/e3b3e264-15d7-479d-9937-2a74499c9c89)

null, does not appear anywhere in method, no good room for `Optional.ofNullable`.

And yes, we can try to refactor method more in next PR to make method to be fashioned chain-methods to make it be similar to any other logic around a world, but it will be fashion update.